src/gl_context.rs

```diff
@@ -26,22 +32,25 @@ pub struct GLContext<Native> {
 }
 
 impl<Native> GLContext<Native>
-    where Native: NativeGLContextMethods
+    where Native: NativeGLContextMethods,
 {
-    pub fn create(shared_with: Option<&Native::Handle>) -> Result<GLContext<Native>, &'static str> {
+    pub fn create(shared_with: Option<&Native::Handle>) -> Result<Self, &'static str> {
         Self::create_shared_with_dispatcher(shared_with, None)
     }
 
     pub fn create_shared_with_dispatcher(shared_with: Option<&Native::Handle>,
                                          dispatcher: Option<Box<GLContextDispatcher>>)
-        -> Result<GLContext<Native>, &'static str> {
+        -> Result<Self, &'static str> {
         let native_context = try!(Native::create_shared_with_dispatcher(shared_with, dispatcher));
+        // XXX how to handle GLES instantiation?
+        let gl_ = gl::GlFns::load_with(|s| GLContext::<Native>::get_proc_address(s) as *const _);
```

Comment:
Oh, right, I just grabbed that from the original Sotaro's patch. I guess the idea here would be doing so per-backend (that is, the egl back-end would load gles functions, etc). That being said, in which case would it make a difference? I guess we can provide only the GLES-like interface given this is primarily for WebGL and we create compat contexts, alternatively, I can make it generic, but that cries to avoid the virtual dispatch then.